rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -164,28 +194,25 @@ def _recursively_initialize_decoders(
                         )
                         continue
 
-                    address_results.update(self.decoders[class_name].addresses_to_decoders())
-                    rules_results.extend(self.decoders[class_name].decoding_rules())
-                    enricher_results.extend(self.decoders[class_name].enricher_rules())
-                    self.all_counterparties.update(self.decoders[class_name].counterparties())
+                    results.address_mappings.update(self.decoders[class_name].addresses_to_decoders())  # noqa: E501
+                    results.event_rules.extend(self.decoders[class_name].decoding_rules())
+                    results.token_enricher_rules.extend(self.decoders[class_name].enricher_rules())
+                    results.post_decoding_rules.extend(self.decoders[class_name].post_decoding_rules())  # noqa: E501
+                    results.counterparties.update(self.decoders[class_name].counterparties())
 
-                recursive_addrs, recursive_rules, recurisve_enricher_results = self._recursively_initialize_decoders(full_name)  # noqa: E501
-                address_results.update(recursive_addrs)
-                rules_results.extend(recursive_rules)
-                enricher_results.extend(recurisve_enricher_results)
+                recursive_results = self._recursively_initialize_decoders(full_name)
+                results += recursive_results
 
-        return address_results, rules_results, enricher_results
+        results.post_decoding_rules.sort(key=lambda x: x[1], reverse=True)
```

Comment:
I know what this is, but we will both forget in a week. Explain why this is sorting and what it's sorting on